repository/src/interpret/dms.rs

```diff
@@ -43,20 +48,233 @@ impl DmsMessage for Message {
+                // commit_message에서 vec<PayloadCommit>을 한번에 보낼 수 없다면, commit들을 하나하나 보낸 다음에 peer가 그거 받아서 재조립해야되는데 씹재앙
+                // Message Type으로 만들어서 한번에 보내는건 안되나?
+            }
+        } else if branch.starts_with("b-") {
+            let b_commit_hash = _raw.write().await.locate_branch(branch.clone()).await?;
+            let commit_hashes = _raw
+                .write()
+                .await
+                .query_commit_path(b_commit_hash, last_finalized_commit_hash)
+                .await?;
+            let payload_branch = PayloadBranch {
+                branch_type: BranchType::Block,
+                commit_hashes: commit_hashes.clone(),
+                branch_name: branch.clone(),
+            };
+            let message = Message::Branch(payload_branch);
+            _dms.write().await.commit_message(&message).await?;
+
+            let mut prev_commit = last_finalized_commit_hash;
+            for (_index, commit_hash) in commit_hashes.iter().enumerate() {
+                let payload_commit = PayloadCommit {
+                    commit: _raw.write().await.read_commit(*commit_hash).await?,
+                    hash: *commit_hash,
+                    parent_hash: prev_commit,
+                    branch_name: branch.clone(),
+                    index: _index,
+                };
+                prev_commit = *commit_hash;
+
+                let message = Message::Commit(payload_commit);
+                _dms.write().await.commit_message(&message).await?;
+            }
+        } else {
+            continue;
+        }
+    }
+
+    // filtering을 해서 a-~ b-~ 이 두 가지 경우를 제외하고는 아무것도 안하고 그 두 경우에는 encoding해서 send한다.
+
     todo!()
 }
 
 /// Updates the repository module with the latest messages from the DMS.
 ///
 /// Note that it never finalizes a block.
 /// Finalization is done by the consensus module, or the `sync` method.
+/*
+   예외처리 & Detail
+   1. (First Commit hash of received things) == (local repo's final hash)
+   2. payload commit들 순서 보장하는거 -> 근데 걍 이거 Message type을 하나 더 만드는게 나을 것 같은데
+   3. PayloadBranch가 아직 도착하지 않았는데 PayloadCommit vector들이 먼저 도착 -> 순서가 올 때까지 기다려야함
+       -> commit hash값으로 알 수 있을 것 같음
+   4.
+*/
+
+// 근데 걍 PayloadCommitSequence를 보내면 안되나?
 pub async fn update(
     _raw: Arc<RwLock<RawRepository>>,
     _dms: Arc<RwLock<Dms<Message>>>,
+    _commit_buffer: &mut std::collections::HashMap<String, Vec<PayloadCommit>>,
 ) -> Result<(), Error> {
+    let messages = _dms.write().await.read_messages().await?;
+    let last_message = messages.last().unwrap().clone().message;
+
+    // 확인해야 할 부분 :
+    // semantic commit을 만들어서 raw에 업데이트 하는 과정을 for문 안에서 create_semantic_commit을 하는 식으로 처리했는데,
+    // 이렇게 하는 방식이 맞는가?
+    match last_message {
+        Message::Branch(payloadbranch) => {
+            let commit_hashes = payloadbranch.commit_hashes; // sender가 보낸 commit hash들
+            let branch_name = payloadbranch.branch_name; // sender의 branch-name을 그대로 이용
```

Comment:
You should not. The branch name must be **deduced from the content** (you know that there is a deterministic rule that decides the name of branch)